frontend/projects/upgrade/src/app/features/dashboard/segments/components/modal/import-segment/import-segment.component.ts

```diff
@@ -30,31 +44,61 @@ export class ImportSegmentComponent {
 
   importSegment() {
     // TODO: improve the logic here
-    const userIds = this.segmentInfo.individualForSegment.map((individual) => individual.userId);
-    const subSegmentIds = this.segmentInfo.subSegments.map((subSegment) => subSegment.id);
-    const groups = this.segmentInfo.groupForSegment.map((group) => ({ type: group.type, groupId: group.groupId }));
-
-    this.segmentTemp = { ...this.segmentInfo, userIds: userIds, subSegmentIds: subSegmentIds, groups: groups };
-    this.isSegmentJSONValid = this.validateSegmentJSON(this.segmentTemp);
-    if (this.isSegmentJSONValid) {
-      this.segmentsService.importSegment({ ...this.segmentTemp });
+    if (this.importFileErrors.length === 0) {
+      this.segmentsService.importSegment(this.allSegments);
       this.onCancelClick();
     }
   }
 
   uploadFile(event) {
+    let index = 0;
+    let fileName = '';
     const reader = new FileReader();
+    this.uploadedFileCount = event.target.files.length;
+    this.importFileErrors = [];
+
     reader.addEventListener(
       'load',
-      function () {
+      async function () {
         const result = JSON.parse(reader.result as any);
         this.segmentInfo = result;
+        this.importFileErrorsDataSource.data = await this.validateSegment(this.segmentInfo, fileName);
+        readFile(++index);
       }.bind(this)
     );
-    reader.readAsText(event.target.files[0]);
+
+    readFile(index);
+    function readFile(fileIndex: number) {
+      if (fileIndex >= event.target.files.length) return;
+      fileName = event.target.files[fileIndex].name;
+      reader.readAsText(event.target.files[fileIndex]);
+    }
+  }
+
+  async validateSegment(segmentInfo: Segment, fileName) {
+    const userIds = segmentInfo.individualForSegment.map((individual) =>
```

Comment:
map can return undefined in this code do you want that?

Reply:
resolved